kcalcore/icaltimezones.cpp

```diff
@@ -1313,10 +1313,10 @@ QList<QDateTime> ICalTimeZoneSourcePrivate::parsePhase( icalcomponent *c,
   }
 
   // Convert DTSTART to QDateTime, and from local time to UTC
-  const QDateTime localStart = toQDateTime( dtstart );   // local time
   dtstart.second -= prevOffset;
   dtstart.is_utc = 1;
   const QDateTime utcStart = toQDateTime( icaltime_normalize( dtstart ) );   // UTC
+  const QDateTime localStart = utcStart.toLocalTime();  // local time
```

Comment:
This fixes testTwo() from the test app, for me.  Well, it fixed testTwo() for me without the second change applied already... I wonder if this change is unnecessary if we just apply the second change (ie, maybe the second change fixes all issues, even without this change?)  I'll test this today.

Regarding libical upgrade - this is a nice idea, as I believe that the current version of libical we use has a memory leak in it, too, related to timezones IIRC.  But I haven't tested.  I don't have write access to mer-core/libical.git though, but I will try updating to latest upstream in a local branch and see if updating helps, also.